File: lab_5/Lab_5/Series/AriProg.cpp
#include "AriProg.hpp"

AriProg::AriProg(double first, double step, int size) : Series(first, step, size) {}

AriProg::~AriProg() {
    std::cout << "AriProg::~AriProg()" << std::endl;
}

double AriProg::getElement(int index) {
    double element = this->first + (index - 1) * this->step;
    return element;
}

double AriProg::getSum() {
    double last = (this->first + (this->size - 1) * this->step);
    return (this->first + last) / 2 * this->size;
}

File: lab_5/lab_5(java)/src/Series/AriProg.java
package Series;

public class AriProg extends Series{
    public AriProg(double first, double step, int size) {
        super(first, step, size);
    }

    public AriProg() {
        super(1, 2, 5);
    }

    @Override
    public double getElement(int index) {
        double element = this.first + (index-1) * this.step;
        return element;
    }

    @Override
    public double getSum() {
        double last = (this.first + (this.size - 1) * this.step);
        return (this.first + last) / 2 * this.size;
    }
}
